pages/uploader.tsx

```diff
@@ -1,150 +1,268 @@
+    //   xhr.open('POST', url);
+    //   xhr.onload = () => {
+    //     res('00')
+    //   }
+    //   xhr.onerror = (evt) => rej(evt);
+    //   xhr.upload.onprogress = (event) => {
+    //     if (event.lengthComputable) {
+    //       const percentage = (event.loaded / event.total) * 100;
+    //       onProgress(Math.round(percentage))
+    //     }
+    //   };
+    //   const formData = new FormData();
+    //   formData.append('file', file);
+    //   formData.append('key', key)
+    //   xhr.send(formData)
+    // });
+  }
+
+  return (
+    <div className={styles.container}>
+      <Navbar />
+      <h1>Uploader</h1>
+      <form>
+        <label>
+          Name:
+          <input
+            type="text"
+            value={title}
+            onChange={(e) => setTitle(e.target.value)}
+          />
+        </label>
+        <label>
+          Subtitle:
+          <input
+            type="text"
+            value={subtitle}
+            onChange={(e) => setSubtitle(e.target.value)}
+          />
+        </label>
+        <label>
+          Date:
+          <input
+            type="text"
+            value={date}
+            onChange={(e) => setDate(e.target.value)}
+            placeholder="MM/DD/YYYY"
+          />
+        </label>
+        <label>
+          Description:
+          <input
+            type="text"
+            value={description}
+            onChange={(e) => setDescription(e.target.value)}
+          />
+        </label>
+        <label>
+          Speaker:
+          <input
+            type="text"
+            value={speaker}
+            onChange={(e) => setSpeaker(e.target.value)}
+          />
+        </label>
+        <label>
+          Scripture:
+          <input
+            type="text"
+            value={scripture}
+            onChange={(e) => setScripture(e.target.value)}
+          />
+        </label>
+        <label>
+          Topic:
+          <input
+            type="text"
+            value={topic}
+            onChange={(e) => setTopic(e.target.value)}
+          />
+        </label>
+      </form>
+      <form className={styles.form}>
+        {file ? (
+          <>
+            <AudioTrimmer url={file.preview}></AudioTrimmer>
+            <button
+              type="button"
+              className={styles.button}
+              onClick={() => {
+                setFile(undefined);
+                setUploadProgress(undefined);
+              }}
+            >
+              Clear
+            </button>
+          </>
+        ) : (
+          <div className={styles.dragAndDrop} {...getRootProps()}>
+            <input type="hidden" {...getInputProps()} />
+            <p>
+              Drag &apos;n&apos; drop audio files here, or click to select files
+            </p>
+          </div>
+        )}
+        <input
+          className={styles.button}
+          type="button"
+          value="Upload"
+          disabled={file === undefined}
```

Comment:
There should be specific fields that are required before the upload button is re-enabled. I think there are some helpful things in [form validation](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation). Also here is where a ui package can come into play lets discuss the options with the team